src/api/common/helpers/postgres.js

```diff
@@ -1,53 +1,66 @@
-import knex from 'knex'
-import { newDb } from 'pg-mem'
-import { fileURLToPath } from 'node:url'
-import path from 'node:path'
-
 import { config } from '~/src/config/index.js'
+import pg from 'pg'
+import { Signer } from '@aws-sdk/rds-signer'
+import { fromNodeProviderChain } from '@aws-sdk/credential-providers'
 
+const { Client } = pg
 /**
  * @satisfies { import('@hapi/hapi').ServerRegisterPluginObject<*> }
  */
+export const postgres = {
+  plugin: {
+    name: 'postgres',
+    version: '1.0.0',
+    /**
+     *
+     * @param { import('@hapi/hapi').Server } server
+     * @param { postgres: object, region: string, isDevelopment: boolean } options
+     * @returns {Promise<void>}
+     */
+    register: async function (server, options) {
+      server.logger.info('Setting up Postgres')
 
-const dirname = path.dirname(fileURLToPath(import.meta.url))
+      let token
 
-const migrationsPath = path.resolve(
-  dirname,
-  '..',
-  '..',
-  '..',
-  'knex',
-  'migrations'
-)
+      if (options.isProduction) {
+        const signer = new Signer({
+          hostname: options.postgres.host,
+          port: options.postgres.port,
+          username: options.postgres.user,
+          credentials: fromNodeProviderChain(),
+          region: options.region
+        })
+        token = await signer.getAuthToken()
+      } else {
+        token = 'admin'
+      }
 
-export const postgres =
-  process.env.NODE_ENV === 'test'
-    ? newDb().adapters.createKnex(0, {
-        client: 'pg',
-        migrations: {
-          directory: migrationsPath,
-          loadExtensions: ['.cjs']
-        }
-      })
-    : knex({
-        client: 'pg',
-        connection: {
-          host: config.get('postgres.host'),
-          port: config.get('postgres.port'),
-          user: config.get('postgres.user'),
-          database: config.get('postgres.database'),
-          password: config.get('postgres.password'),
-          ssl: config.get('postgres.ssl')
-            ? { rejectUnauthorized: false }
-            : false
-        },
-        migrations: {
-          directory: migrationsPath,
-          loadExtensions: ['.cjs']
-        }
-      })
+      const db = () =>
+        new Client({
+          user: options.postgres.user,
+          password: token,
+          host: options.postgres.host,
+          port: options.postgres.port,
+          database: options.postgres.database
+        })
 
-/**
- * To be mixed in with Request|Server to provide the db decorator
- * @typedef {{db: import('~/src/api/common/helpers/postgres.js').postgres }}
- */
+      server.logger.info(
+        `Postgres connected to database '${options.postgres.database}'`
+      )
+
+      server.decorate('server', 'db', db)
+      server.decorate('request', 'db', db)
+    }
+  },
+  options: {
+    postgres: config.get('postgres'),
+    region: 'eu-west-2',
+    isDevelopment: config.get('isDevelopment')
```

Comment:
fine for a proof of concept but ideally this sort of stuff should be bound to an named config setting (postgres.useRDS/postgres.authMode or something).